torch/csrc/autograd/variable.h

```diff
@@ -267,8 +267,7 @@ struct TORCH_API AutogradMeta : public c10::AutogradMetaInterface {
   /// Sets the `requires_grad` property of `Variable`. This should be true for
   /// leaf variables that want to accumulate gradients, and false for all other
   /// variables.
-  void set_requires_grad(bool requires_grad, at::TensorImpl* self_impl)
-      override {
+  void set_requires_grad(bool requires_grad, at::TensorImpl* self_impl) final {
```

Comment:
Clang-tidy complains the virtual function is called in the constructor. However, if it is final, then the warning disappear. And it should be final, otherwise there is a logical error.

Reply:
Two fixes, you either make it final like so, or you explicitly have the ctor call this classes' version of set_requires_grad. Not sure which is the proper solution here, especially since overriding it here is only adding a few additional checks before actually setting the value.

Reply:
It just feels weird this is the only method in the class that should be final?

Reply:
Actually, upon further reading, it's all internal APIs so should be fine.

Reply:
@Skylion007  I tried to use AutogradMeta::set_requires_grad but later decided to use 'final' so that further developer will notice the danger before they remove 'final' and override the method.